internal/auth/oidc/service_callback.go

```diff
@@ -46,7 +46,7 @@ func Callback(
 	oidcRepoFn OidcRepoFactory,
 	iamRepoFn IamRepoFactory,
 	atRepoFn AuthTokenRepoFactory,
-	authMethodId,
+	am *AuthMethod,
```

Comment:
why pass the entire auth method, when just its public id is needed?

Reply:
it would appear all the changes in this method and the tests aren't needed if we keep the more precise parameter of an auth method id.